File: examples/sample/UserService.java
package sample;

import java.util.Optional;

public class UserService {
    private final UserRepository repository;

    public UserService(UserRepository repository) {
        this.repository = repository;
    }

    public Optional<User> findById(String id) {
        return repository.loadUser(id).map(this::hydrate);
    }

    private User hydrate(User user) {
        return user.withTrimmedName();
    }

    public String renderUser(String id) {
        return findById(id).map(User::display).orElse("(missing)");
    }
}

File: examples/sample/UserService.cs
using System;
using System.Collections.Generic;

namespace Sample
{
    public class UserService
    {
        public string FindById(int id)
        {
            return Render(LoadUser(id));
        }

        private string LoadUser(int id)
        {
            return $"user#{id}";
        }

        private string Render(string user)
        {
            return $"[{user}]";
        }

        public string RenderUser(int id)
        {
            return FindById(id);
        }
    }
}
